### PR TITLE
Add conversion from dSpacing to ToF

### PR DESCRIPTION
**Pull request by DanNixon**
Tests using a round trip conversion from ToF to dSpacing and back to ToF.

Adds helper function to compare values with a tolerance.
This was needed sue to a small amount of floating point error when performing the round trip conversions with sparse data (ToF).

Re #267

### CODE REVIEW

File: neutron/test/convert_test.cpp [hunk 1]
```diff
@@ -144,6 +144,34 @@ TEST(Convert, Tof_to_DSpacing) {
             tof.labels()["component_info"]);
 }
 
+TEST(Convert, DSpacing_to_Tof) {
+  /* Assuming the Tof_to_DSpacing test is correct and passing we can test the
+   * inverse conversion by simply comparing a round trip conversion with the
+   * original data. */
```
Comment (SimonHeybrock) [inline neutron/test/convert_test.cpp]: 👍 good way of testing.

File: neutron/test/convert_test.cpp [hunk 2]
```diff
@@ -144,6 +144,34 @@ TEST(Convert, Tof_to_DSpacing) {
...
+
+  const Dataset tof_original = makeTofDataForUnitConversion();
+  const auto dspacing = convert(tof_original, Dim::Tof, Dim::DSpacing);
+  const auto tof = convert(dspacing, Dim::DSpacing, Dim::Tof);
+
+  /* Test coordinates */
+  /* Slicing is needed as conversion introduces the dependance on Dim::Position
+   */
+  EXPECT_EQ(tof.coords()[Dim::Tof].slice({Dim::Position, 0}),
+            tof_original.coords()[Dim::Tof]);
```
Comment (SimonHeybrock) [inline neutron/test/convert_test.cpp]: Shouldn't you test all slices? Or use `broadcast` to bring the reference to the same shape?

**Comment by DanNixon** [comment]
> The inexact comparison may take a while to implement?

Not too sure what you are referring to here.

**Comment by SimonHeybrock** [comment]
> > The inexact comparison may take a while to implement?
> 
> Not too sure what you are referring to here.

If was thinking of a generic implementation that would just let you do `similar(dataset1, dataset2, tolerance=1e-12, allow_transpose=True, allow_broadcast=True)`, or something like that.

**Pull request merged**